stacker_blueprints/empire/daemon.py

```diff
@@ -0,0 +1,635 @@
+        "ServiceMaximumPercent": {
+            "type": "Number",
+            "description": (
+                "The maximum number of tasks, specified as a percentage of the "
+                "Amazon ECS service's DesiredCount value, that can run in a "
+                "service during a deployment."
+            ),
+            "default": "200"},
+        "ServiceMinimumHealthyPercent": {
+            "type": "Number",
+            "description": (
+                "The minimum number of tasks, specified as a percentage of the "
+                "Amazon ECS service's DesiredCount value, that must continue "
+                "to run and remain healthy during a deployment."
+            ),
+            "default": "50"}
+    }
+
+    def create_template(self):
+        self.create_conditions()
+        self.create_security_groups()
+        self.create_custom_cloudformation_resources()
+        self.create_template_bucket()
+        self.create_load_balancer()
+        self.create_ecs_resources()
+        self.create_log_group()
+
+    def create_conditions(self):
+        t = self.template
+        ssl_condition = Not(Equals(Ref("ELBCertName"), ""))
+        t.add_condition("UseSSL", ssl_condition)
+        t.add_condition("UseHTTP", Not(ssl_condition))
+        self.template.add_condition(
+            "UseIAMCert",
+            Not(Equals(Ref("ELBCertType"), "acm")))
+        t.add_condition(
+            "EnableSNSEvents",
+            Equals(Ref("EventsBackend"), "sns"))
+        t.add_condition(
+            "CreateSNSTopic",
+            And(Equals(Ref("EventsSNSTopicName"), ""),
+                Condition("EnableSNSEvents")))
+        t.add_condition(
+            "EnableCloudwatchLogs",
+            Equals(Ref("RunLogsBackend"), "cloudwatch"))
+        t.add_condition(
+            "CreateRunLogsGroup",
+            And(Equals(Ref("RunLogsCloudwatchGroup"), ""),
+                Condition("EnableCloudwatchLogs")))
+        t.add_condition(
+            "EnableAppEventStream",
+            Equals(Ref("LogsStreamer"), "kinesis"))
+
+    def create_security_groups(self):
+        t = self.template
+
+        t.add_resource(
+            ec2.SecurityGroup(
+                ELB_SG_NAME,
+                GroupDescription="Security group for load balancer",
+                VpcId=Ref("VpcId")))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "ELBPort80FromTrustedNetwork",
+                IpProtocol="tcp", FromPort="80", ToPort="80",
+                CidrIp=Ref("TrustedNetwork"),
+                Condition="UseHTTP",
+                GroupId=Ref(ELB_SG_NAME)))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "ELBPort443FromTrustedNetwork",
+                IpProtocol="tcp", FromPort="443", ToPort="443",
+                CidrIp=Ref("TrustedNetwork"),
+                Condition="UseSSL",
+                GroupId=Ref(ELB_SG_NAME)))
+        t.add_resource(
```

Comment:
This only runs if HTTPS is enabled - seems like the right thing to do, but might be worth documenting somewhere.